--- origsrc/xapian-bindings-1.0.7/java/native/Makefile.am	2008-07-15 23:42:40.000000000 -0500
+++ src/xapian-bindings-1.0.7/java/native/Makefile.am	2008-07-16 21:23:21.546875000 -0500
@@ -37,5 +37,5 @@
 endif
 AM_CXXFLAGS = $(JAVA_CXXFLAGS) $(XAPIAN_CXXFLAGS)
 AM_CPPFLAGS = $(JAVA_CPPFLAGS)
-libxapian_jni_la_LDFLAGS = -avoid-version -module $(SHREXT) $(NO_UNDEFINED)
+libxapian_jni_la_LDFLAGS = -avoid-version $(SHREXT) $(NO_UNDEFINED)
 libxapian_jni_la_LIBADD = $(XAPIAN_LIBS)
--- origsrc/xapian-bindings-1.0.7/ruby/Makefile.am	2008-07-15 23:42:42.000000000 -0500
+++ src/xapian-bindings-1.0.7/ruby/Makefile.am	2008-07-16 21:25:07.531250000 -0500
