--- origsrc/xapian-bindings-1.0.7/configure.ac	2008-07-15 23:42:44.000000000 -0500
+++ src/xapian-bindings-1.0.7/configure.ac	2008-07-16 21:49:18.937500000 -0500
@@ -440,6 +440,14 @@
 	fi
       fi
       AC_SUBST(TCL_CPPFLAGS)
+	case $host_os in
+	cygwin*)
+		TCL_VER=`echo 'puts $tcl_version'|$TCLSH`
+		AC_CHECK_LIB(tcl$TCL_VER, Tcl_Init, TCL_LIBS="-ltcl$TCL_VER",
+				[AC_MSG_ERROR([libtcl not found])])
+		;;
+	esac
+	AC_SUBST(TCL_LIBS)
     else
       AC_MSG_RESULT([< 8.1 (too old)])
       if test yes = "$with_tcl" ; then
--- origsrc/xapian-bindings-1.0.7/tcl8/Makefile.am	2008-07-15 23:42:44.000000000 -0500
+++ src/xapian-bindings-1.0.7/tcl8/Makefile.am	2008-07-16 21:27:20.859375000 -0500
@@ -28,11 +28,11 @@
 	eval `grep '^dlname=' $(tcllib_LTLIBRARIES)` ; \
 	  rm -f $(DESTDIR)$(tcllibdir)/"$$dlname"
 
-AM_CPPFLAGS = @TCL_CPPFLAGS@ -DUSE_TCL_STUBS
+AM_CPPFLAGS = @TCL_CPPFLAGS@
 AM_CXXFLAGS = @SWIG_CXXFLAGS@ $(XAPIAN_CXXFLAGS)
 xapian_la_LDFLAGS = -avoid-version -module $(NO_UNDEFINED)
 nodist_xapian_la_SOURCES = xapian_wrap.cc
-xapian_la_LIBADD = $(XAPIAN_LIBS)
+xapian_la_LIBADD = $(XAPIAN_LIBS) $(TCL_LIBS)
 
 BUILT_SOURCES = xapian_wrap.cc
 CLEANFILES = pkgIndex.tcl
