--- origsrc/xapian-bindings-1.0.7/csharp/Makefile.am	2008-07-15 23:42:40.000000000 -0500
+++ src/xapian-bindings-1.0.7/csharp/Makefile.am	2008-07-16 21:33:31.375000000 -0500
@@ -67,12 +67,12 @@
 
 EXTRA_DIST = util.i AssemblyInfo.cs.in SmokeTest.cs $(BUILT_SOURCES)
 
-lib_LTLIBRARIES = XapianSharp.la
+lib_LTLIBRARIES = libXapianSharp.la
 
 AM_CXXFLAGS = @SWIG_CXXFLAGS@ $(XAPIAN_CXXFLAGS)
-XapianSharp_la_LDFLAGS = -avoid-version -module $(NO_UNDEFINED)
-XapianSharp_la_SOURCES = xapian_wrap.cc
-XapianSharp_la_LIBADD = $(XAPIAN_LIBS)
+libXapianSharp_la_LDFLAGS = -avoid-version $(NO_UNDEFINED)
+libXapianSharp_la_SOURCES = xapian_wrap.cc
+libXapianSharp_la_LIBADD = $(XAPIAN_LIBS)
 
 install-data-local: $(ASSEMBLY).dll
 	$(GACUTIL) /i $(ASSEMBLY).dll /f /package $(ASSEMBLY) /gacdir $(libdir) /root $(DESTDIR)$(libdir)
