--- origsrc/xapian-bindings-1.0.7/ruby/Makefile.am	2008-07-15 23:42:42.000000000 -0500
+++ src/xapian-bindings-1.0.7/ruby/Makefile.am	2008-07-16 21:25:07.531250000 -0500
@@ -34,7 +34,7 @@
 AM_CXXFLAGS = @SWIG_CXXFLAGS@ $(XAPIAN_CXXFLAGS)
 _xapian_la_LDFLAGS = -avoid-version -module -shrext ".$(RUBY_DLEXT)" $(NO_UNDEFINED)
 _xapian_la_SOURCES = xapian_wrap.cc xapian_wrap.h
-_xapian_la_LIBADD = $(XAPIAN_LIBS)
+_xapian_la_LIBADD = $(XAPIAN_LIBS) -lruby
 
 if MAINTAINER_MODE
 xapian_wrap.cc xapian_wrap.h: xapian_wrap.stamp
