--- origsrc/xapian-bindings-1.2.12/configure.ac	2012-08-23 23:37:20.529891100 -0500
+++ src/xapian-bindings-1.2.12/configure.ac	2012-08-24 00:07:27.846263700 -0500
@@ -836,7 +836,7 @@ if test no != "$with_ruby" ; then
 
 	RUBY_LIBS=
 	case $host_os in
-	cygwin*) RUBY_LIBS=-lruby ;;
+	cygwin*) [RUBY_LIBS=`$RUBY -rrbconfig -e 'puts RbConfig::CONFIG["LIBRUBYARG"]'`] ;;
 	esac
 	AC_SUBST(RUBY_LIBS)
 
