--- origsrc/xapian-bindings-1.2.12/configure.ac	2012-06-27 08:08:28.000000000 -0500
+++ src/xapian-bindings-1.2.12/configure.ac	2012-07-31 22:48:49.216551300 -0500
@@ -409,7 +409,7 @@ if test no != "$with_php" ; then
 	dnl there.  FIXME.
 	PHP_LIBS="-L`$PHP_CONFIG --prefix` -lphp${PHP_MAJOR_VERSION}ts" ;;
       cygwin*)
-	PHP_LIBS="-lphp${PHP_MAJOR_VERSION}lib" ;;
+	PHP_LIBS="-lphp${PHP_MAJOR_VERSION}" ;;
       *)
 	PHP_LIBS= ;;
       esac
