--- origsrc/xapian-bindings-1.2.12/configure.ac	2012-06-27 08:08:28.000000000 -0500
+++ src/xapian-bindings-1.2.12/configure.ac	2012-07-31 22:48:49.216551300 -0500
@@ -471,17 +471,8 @@ if test no != "$with_tcl" ; then
 	fi
       fi
 
-      dnl Cygwin needs to link with -ltcl8.5 or similar.
-      TCL_LIBS=
-      case $host_os in
-      cygwin*)
-	AC_CHECK_LIB([tcl$tcl_version], [Tcl_Init],
-		     [TCL_LIBS="-ltcl$tcl_version"],
-		     [AC_MSG_ERROR([libtcl$tcl_version not found])])
-	;;
-      *)
-	TCL_CPPFLAGS="$TCL_CPPFLAGS -DUSE_TCL_STUBS" ;;
-      esac
+      TCL_CPPFLAGS="$TCL_CPPFLAGS -DUSE_TCL_STUBS"
+      TCL_LIBS="-Wl,-ltclstub$tcl_version"
       AC_SUBST(TCL_CPPFLAGS)
       AC_SUBST(TCL_LIBS)
     else
