--- origsrc/xapian-bindings-1.0.7/configure.ac	2008-07-15 23:42:44.000000000 -0500
+++ src/xapian-bindings-1.0.7/configure.ac	2008-07-16 21:49:18.937500000 -0500
@@ -54,6 +54,8 @@
 
 dnl Use libtool to manage our libraries, but don't build static libraries as
 dnl the bindings only have a use for dynamic ones.
+AC_PROG_CC
+AC_PROG_CXX
 AC_DISABLE_STATIC
 AC_PROG_LIBTOOL
 
@@ -252,6 +254,8 @@
 	case $host_os in
 	mingw* | pw32*)
 	  PYTHON_LIBS=`$PYTHON -c 'import os,sys;print("-L"+os.path.join(sys.prefix,"libs").replace(os.sep,"/")+" -lpython"+sys.version[[:3]].replace(".",""))'` ;;
+	cygwin*)
+	  PYTHON_LIBS=`$PYTHON -c 'import os,sys;print "-L"+os.path.join(sys.path[[3]],"config"),"-lpython"+sys.version[[:3]]'` ;;
 	*)
 	  PYTHON_LIBS= ;;
 	esac
@@ -373,6 +377,8 @@
 	dnl someone who actually has mingw and wants to build the PHP bindings
 	dnl there.  FIXME.
 	PHP_LIBS="-L`$PHP_CONFIG --prefix` -lphp${PHP_MAJOR_VERSION}ts" ;;
+      cygwin*)
+	PHP_LIBS="-lphp${PHP_MAJOR_VERSION}lib" ;;
       *)
 	PHP_LIBS= ;;
       esac
@@ -434,6 +440,14 @@
 	fi
       fi
       AC_SUBST(TCL_CPPFLAGS)
+	case $host_os in
+	cygwin*)
+		TCL_VER=`echo 'puts $tcl_version'|$TCLSH`
+		AC_CHECK_LIB(tcl$TCL_VER, Tcl_Init, TCL_LIBS="-ltcl$TCL_VER",
+				[AC_MSG_ERROR([libtcl not found])])
+		;;
+	esac
+	AC_SUBST(TCL_LIBS)
     else
       AC_MSG_RESULT([< 8.1 (too old)])
       if test yes = "$with_tcl" ; then
--- origsrc/xapian-bindings-1.0.7/csharp/Makefile.am	2008-07-15 23:42:40.000000000 -0500
+++ src/xapian-bindings-1.0.7/csharp/Makefile.am	2008-07-16 21:33:31.375000000 -0500
@@ -67,12 +67,12 @@
 
 EXTRA_DIST = util.i AssemblyInfo.cs.in SmokeTest.cs $(BUILT_SOURCES)
 
-lib_LTLIBRARIES = XapianSharp.la
+lib_LTLIBRARIES = libXapianSharp.la
 
 AM_CXXFLAGS = @SWIG_CXXFLAGS@ $(XAPIAN_CXXFLAGS)
-XapianSharp_la_LDFLAGS = -avoid-version -module $(NO_UNDEFINED)
-XapianSharp_la_SOURCES = xapian_wrap.cc
-XapianSharp_la_LIBADD = $(XAPIAN_LIBS)
+libXapianSharp_la_LDFLAGS = -avoid-version $(NO_UNDEFINED)
+libXapianSharp_la_SOURCES = xapian_wrap.cc
+libXapianSharp_la_LIBADD = $(XAPIAN_LIBS)
 
 install-data-local: $(ASSEMBLY).dll
 	$(GACUTIL) /i $(ASSEMBLY).dll /f /package $(ASSEMBLY) /gacdir $(libdir) /root $(DESTDIR)$(libdir)
--- origsrc/xapian-bindings-1.0.7/java/native/Makefile.am	2008-07-15 23:42:40.000000000 -0500
+++ src/xapian-bindings-1.0.7/java/native/Makefile.am	2008-07-16 21:23:21.546875000 -0500
@@ -37,5 +37,5 @@
 endif
 AM_CXXFLAGS = $(JAVA_CXXFLAGS) $(XAPIAN_CXXFLAGS)
 AM_CPPFLAGS = $(JAVA_CPPFLAGS)
-libxapian_jni_la_LDFLAGS = -avoid-version -module $(SHREXT) $(NO_UNDEFINED)
+libxapian_jni_la_LDFLAGS = -avoid-version $(SHREXT) $(NO_UNDEFINED)
 libxapian_jni_la_LIBADD = $(XAPIAN_LIBS)
--- origsrc/xapian-bindings-1.0.7/python/Makefile.am	2008-07-15 23:42:41.000000000 -0500
+++ src/xapian-bindings-1.0.7/python/Makefile.am	2008-07-16 21:27:20.843750000 -0500
@@ -23,7 +23,7 @@
 
 # Install as _DATA rather than _SCRIPTS because we don't want to make these
 # executable (they don't have a #! line).
-pylib_DATA = xapian.py xapian.pyc xapian.pyo
+pylib_DATA = xapian.py
 
 pylib_LTLIBRARIES = _xapian.la
 
--- origsrc/xapian-bindings-1.0.7/ruby/Makefile.am	2008-07-15 23:42:42.000000000 -0500
+++ src/xapian-bindings-1.0.7/ruby/Makefile.am	2008-07-16 21:25:07.531250000 -0500
@@ -34,7 +34,7 @@
 AM_CXXFLAGS = @SWIG_CXXFLAGS@ $(XAPIAN_CXXFLAGS)
 _xapian_la_LDFLAGS = -avoid-version -module -shrext ".$(RUBY_DLEXT)" $(NO_UNDEFINED)
 _xapian_la_SOURCES = xapian_wrap.cc xapian_wrap.h
-_xapian_la_LIBADD = $(XAPIAN_LIBS)
+_xapian_la_LIBADD = $(XAPIAN_LIBS) -lruby
 
 if MAINTAINER_MODE
 xapian_wrap.cc xapian_wrap.h: xapian_wrap.stamp
--- origsrc/xapian-bindings-1.0.7/tcl8/Makefile.am	2008-07-15 23:42:44.000000000 -0500
+++ src/xapian-bindings-1.0.7/tcl8/Makefile.am	2008-07-16 21:27:20.859375000 -0500
@@ -28,11 +28,11 @@
 	eval `grep '^dlname=' $(tcllib_LTLIBRARIES)` ; \
 	  rm -f $(DESTDIR)$(tcllibdir)/"$$dlname"
 
-AM_CPPFLAGS = @TCL_CPPFLAGS@ -DUSE_TCL_STUBS
+AM_CPPFLAGS = @TCL_CPPFLAGS@
 AM_CXXFLAGS = @SWIG_CXXFLAGS@ $(XAPIAN_CXXFLAGS)
 xapian_la_LDFLAGS = -avoid-version -module $(NO_UNDEFINED)
 nodist_xapian_la_SOURCES = xapian_wrap.cc
-xapian_la_LIBADD = $(XAPIAN_LIBS)
+xapian_la_LIBADD = $(XAPIAN_LIBS) $(TCL_LIBS)
 
 BUILT_SOURCES = xapian_wrap.cc
 CLEANFILES = pkgIndex.tcl
